### ISSUE TITLE
Make codebase consistent with regards to providing random key vs random seed

### ISSUE THREAD

**Issue by ghost** [gchq/coreax#332]
In some parts of the codebase, the user supplies a random key (an array of size (2,)). In other parts, the user supplies an integer seed for the random key. Change the codebase to consistently use only one of these approaches.

See [how random works in Jax](https://jax.readthedocs.io/en/latest/jax-101/05-random-numbers.html).

Note that key and seed may have been used interchangeably erroneously.
